zebra-network/src/peer_set/initialize.rs

```diff
@@ -114,16 +117,50 @@ where
         inv_receiver,
     );
     let peer_set = Buffer::new(BoxService::new(peer_set), constants::PEERSET_BUFFER_SIZE);
+    info!("checkpoint: {}", line!());
+
+    // 2. Incoming peer connections, via a listener.
 
-    // Connect the tx end to the 3 peer sources:
+    // Warn if we're configured using the wrong network port.
+    // TODO: use the right port if the port is unspecified
+    //       split the address and port configs?
+    let (wrong_net, wrong_net_port) = match config.network {
+        Network::Mainnet => (Network::Testnet, 18233),
+        Network::Testnet => (Network::Mainnet, 8233),
+    };
+    if config.listen_addr.port() == wrong_net_port {
+        warn!(
+            "We are configured with port {} for {:?}, but that port is the default port for {:?}",
+            config.listen_addr.port(),
+            config.network,
+            wrong_net
+        );
+    }
+
+    info!("checkpoint: {}", line!());
+    let listen_guard = tokio::spawn(listen(config.listen_addr, listener, peerset_tx.clone()));
+    info!("checkpoint: {}", line!());
+
+    let initial_peers_fut = {
+        let initial_peers = config.initial_peers();
+        info!("checkpoint: {}", line!());
+        let connector = connector.clone();
+        info!("checkpoint: {}", line!());
+        let tx = peerset_tx.clone();
+        info!("checkpoint: {}", line!());
+
+        // Connect the tx end to the 3 peer sources:
+        add_initial_peers(initial_peers, connector, tx)
+    };
+
+    info!("checkpoint: {}", line!());
 
     // 1. Initial peers, specified in the config.
-    let add_guard = tokio::spawn(add_initial_peers(
-        config.initial_peers(),
-        connector.clone(),
-        peerset_tx.clone(),
-    ));
+    let add_guard = tokio::spawn(initial_peers_fut);
 
+    info!("checkpoint: {}", line!());
+
+    /*
```

Comment:
Git saves old versions of the code, so you don't need to comment it out.

And keeping commented-out code makes it harder to review your PR.